app/controllers/cart_items_controller.rb

```diff
@@ -6,7 +6,9 @@ class CartItemsController < ApplicationController
   before_action :set_cart_item, only: %i[update destroy]
 
   def create
-    if @cart.save
+    @cart_item = current_user.cart.cart_items.new(cart_item_params)
+
+    if @cart_item.check_age_restriction && @cart_item.save
```

Comment:
check_age_restriction method not required to call from here. the save method automatically triggers validations. check the model callbacks for reference